consensus/src/model/stores/reachability.rs

```diff
@@ -58,19 +59,27 @@ pub struct DbReachabilityStore {
     db: Arc<DB>,
     access: CachedDbAccess<Hash, Arc<ReachabilityData>, BlockHasher>,
     reindex_root: CachedDbItem<Hash>,
+    level: Option<BlockLevel>,
 }
 
 impl DbReachabilityStore {
-    pub fn new(db: Arc<DB>, cache_size: u64) -> Self {
+    pub fn new(db: Arc<DB>, level: Option<BlockLevel>, cache_size: u64) -> Self {
```

Comment:
The `BlockLevel` arg might be a bit misleading (giving the impression that generally reachability is stored per level). Since this is only a specific use case for the pruning proof, perhaps make the arg `prefix_ending: &[u8]` and parse the block level by the caller. Also maybe avoid changing the `new` function and add another ctor function `with_prefix_ending` (which new will call with an empty slice).